### ISSUE TITLE
Take in account DevTool to false

### ISSUE THREAD

**Issue by PierBJX** [webpack-contrib/source-map-loader#180]
### Feature Proposal
I notice that even if `devtool` is not defined or set to false, `source-map-loading` is doing some stuff. I was wondering if it could make sense to deactivate this plugin by default in this case ?

### Feature Use Case
In development mode, you have `devtool` defines to `inline-source-map` (for example) and you need `SourceMapLoader`. However, in production mode, you do not have `devtool` because you don't want any source map. Instead, of having two webpack config, one for prod and one for dev, we could simply deactivate `SourceMapLoader` if there is no `devtool`.

**Comment by alexander-akait** [comment]
Yeah, makes sense, but it still will run loader (so you get extra calls), anyway you can use `.concat` when you setup loaders and don't use this loader when your are in production mode

**Comment by PierBJX** [comment]
- So you mean that the loader will be run just to check the value of the `devtool` property ?
- What does source map loader do if `devtool` is set to `false` ?

**Comment by alexander-akait** [comment]
> So you mean that the loader will be run just to check the value of the devtool property ?

Yes, but it is still extra calls. We avoid using devtool values, because some developer can set `devtool: false`, but load source maps for some other files and reuse it in other loaders/plugins. Due to this we avoid change logic loaders based on some global options, shorty - we don't know how a loader will be used.

> What does source map loader do if devtool is set to false ?

Do the same as with any other value
